### PR TITLE
[MatrixCamp] Add Built-in function lpad

### PR DESCRIPTION
**Pull request by JackTan25**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #1984 

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Comment by domingozhang** [comment]
@JackTan25 
In MySQL, if the len of LPAD(str, len, padstr) is less than 0 such as -1, the function returns NULL. In your code, I find the return is "The second argument of the lpad function must be an int64 constant".
Could you please think about adjusting it as NULL as well?

File: pkg/builtin/multi/lpad.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+	overload.MultiOps[builtin.Lpad] = []*overload.MultiOp{
+		{
+			Min:        3,
+			Max:        3,
+			Typ:        types.T_varchar,
+			ReturnType: types.T_varchar,
+			Fn: func(vecs []*vector.Vector, proc *process.Process, cs []bool) (*vector.Vector, error) {
+				vs := vecs[0].Col.(*types.Bytes) //Get the first arg
+
+				if !cs[1] && vecs[1].Typ.Oid != types.T_int64 {
```
Comment (broccoliSpicy) [inline pkg/builtin/multi/lpad.go:57]: I think the logical operator here is **OR**

**Comment by domingozhang** [comment]
btv passed

File: pkg/builtin/multi/lpad.go (hunk 1 above)
Comment (broccoliSpicy) [reply pkg/builtin/multi/lpad.go:57]: I tried mysql, and I think mysql accepts both constant value and attribute name as lpad's parameter, in that case, we don't need this check
```go
!cs[1]
```

```go
!cs[2]
```

Comment (broccoliSpicy) [reply pkg/builtin/multi/lpad.go:57]: I think we need different function implementations to handle this hairy constant scenario, namely:
 parameter1 | parameter2 | parameter3
 ----------- |  ----------- | ----------
 constant   |    constant.  |   constant
constant.   |    constant.   |  column 
constant.   |    column      | column
...... |


the builtin power function has some neat examples you may refer to

File: pkg/builtin/multi/lpad.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+					vs.Lengths = make([]uint32, len(temp.Lengths))
+					vs.Offsets = make([]uint32, len(temp.Offsets))
+					copy(vs.Data, temp.Data)
+					copy(vs.Lengths, temp.Lengths)
+					copy(vs.Offsets, temp.Offsets)
+					return vecs[0], nil
+				}
+
+				vec, err := process.Get(proc, 24*int64(len(vs.Lengths)), types.Type{Oid: types.T_varchar, Size: 24})
+				if err != nil {
```
Comment (broccoliSpicy) [inline pkg/builtin/multi/lpad.go:108]: I think the size we need for our new vector here is 24, perhaps I am wrong about this

File: pkg/builtin/multi/lpad.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+					return nil, err
+				}
+				nulls.Set(vec.Nsp, vecs[0].Nsp)
+				vector.SetCol(vec, lpad.LpadVarchar(vs, lens, padds))
+				return vec, nil
```
Comment (broccoliSpicy) [inline pkg/builtin/multi/lpad.go:113]: I believe these comments are irrelevant here

File: pkg/builtin/multi/lpad.go (hunk 2 above)
Comment (JackTan25) [reply pkg/builtin/multi/lpad.go:108]: Yeah,We could add WeChat for detail communication more??

**Pull request merged**